File: Python version/languages/pddl/PDDLMapper.py
from languages.Mapper import Mapper
from parsers.pddl.PDDLParser import PDDLParser

class PDDLMapper(Mapper):
    """Contains methods used to transform Objects into InputProgram"""
    __Instance = None

    def __init__(self):
        if PDDLMapper.__Instance:
            raise("Instance already exists")
        super(PDDLMapper, self).__init__()

    @classmethod
    def getInstance(cls):
        """Return the instance of PDDLMapper"""
        if not cls.__Instance:
            cls.__Instance = PDDLMapper()
        return cls.__Instance
    
    def _getActualString(self, predicate, parametersMap):
        return None
    
    def _getId(self, action):
        """Return a string representing a predicate"""
        if action.index('(') != 0:
            raise ("Wrong format")
        
        return action[1:action.index(" ")]
    
    def _getParam(self, action):
        """Return a set of parameter string name"""
        return PDDLParser.parse(action).getParameters()

File: CSharp version/it/unical/mat/embasp/languages/pddl/PDDLMapper.cs
using it.unical.mat.parsers.pddl;
using System;
using System.Collections.Generic;

namespace it.unical.mat.embasp.languages.pddl
{

    using IllegalTermException = it.unical.mat.embasp.languages.asp.IllegalTermException;

    public class PDDLMapper : Mapper
    {
        private static PDDLMapper mapper;

        public static PDDLMapper Instance
        {
            get
            {
                if (PDDLMapper.mapper == null)
                    PDDLMapper.mapper = new PDDLMapper();
                return PDDLMapper.mapper;
            }
        }

        private PDDLMapper() : base() { }

        protected internal override string GetActualString(string predicate, Dictionary<int, object> parametersMap) => null;

        protected internal override string GetId(string @string)
        {
            // I assume that the string is like (zoom plane1 city4 city1 fl4 fl3 fl2)
            int initialB = @string.IndexOf("(");

            if (initialB != 0)
                throw new System.ArgumentException("Wrong format");

            return @string.Substring(1, @string.IndexOf(' ') - 1);
        }

        protected internal override string[] GetParam(string @string) => PDDLParser.Parse(@string).GetParameters();
    }
}